### DISCUSSION TITLE
problem with page route..!

### DISCUSSION THREAD

**Discussion by Diae98** [flet-dev/flet#1709] [Q&A]
this error :TypeError: Control._build_add_commands() missing 1 required positional argument: 'self', when I run this code, idk
where are you, father

`import flet
from flet import *

class Wellcome_page(UserControl):
    def __init__(self):
        super().__init__()

    def go_to_home(self):
        self.page.go('/Home')

    def build(self):
        sec = Container(
                Stack([
                    Image(src="trying expert\purple.jpg"),
                    Container(
                        Container(
                            Column([
                                Text(
                                    "Wellcome",
                                    italic=True,
                                    size=32,
                                    color=colors.WHITE,   
                                ),
                                Text(
                                    "Take breath and see out box",
                                    italic=True,
                                    size=26,
                                    color=colors.WHITE 
                                ),
                                ElevatedButton(
                                    text="Start!",
                                    tooltip="Start use app!",
                                    width=120,
                                    height=70,
                                    color = colors.WHITE,
                                    bgcolor=colors.PURPLE_700,
                                    opacity=0.8,
                                    on_click=lambda e: self.go_to_home(),
                                )

                                ],
                                alignment=MainAxisAlignment.CENTER,
                                horizontal_alignment= CrossAxisAlignment.CENTER    
                            ),
                            width=350,
                        height=200,
                        margin=margin.only(top=150),
                        border_radius=10,
                        blur=Blur(10,12,BlendMode.DARKEN),
                        border=border.all(2,color=colors.PURPLE_800),
                        opacity=0.9
                        ),
                        alignment=alignment.center
                    )
                    ] 
                    )
            )
        return sec
class Home(UserControl):
    def __init__(self):
        super().__init__()
    def build(self):    
        return Text("Yes",size=100)

def main(page:Page):
    page.vertical_alignment = MainAxisAlignment.CENTER
    page.horizontal_alignment = CrossAxisAlignment.CENTER

    wellcome_page = Wellcome_page()
    home = Home
    def route_change(route):
        page.views.clear()
        if page.route == '/':
            page.views.append(wellcome_page)
        if page.route == '/Home':
            page.views.append(home)

    page.on_route_change = route_change
    page.go(page.route)
flet.app(target=main)`

**Comment by FeodorFitsner** [comment]
Please provide complete, runnable code example.
